controllers/mirrorpeer_controller.go

```diff
@@ -236,13 +247,14 @@ func (r *MirrorPeerReconciler) checkForSourceSecret(ctx context.Context, peerRef
 		// Source Namespace for the secret
 		Namespace: peerRef.ClusterName,
 	}, &peerSourceSecret)
+
 	if err != nil {
 		if k8serrors.IsNotFound(err) {
 			logger.Info("Source secret not found", "Secret", prSecretName)
-			return nil
+			return err
 		}
-		return err
```

Comment:
why is this not required? we need to return right?

Reply:
I am returning above.